### PR TITLE
New computation of expectation value for mps method

### PR DESCRIPTION
**Pull request by merav-aharoni**
### Summary
Writing new implementation for computing the expectation value. The new method should be much better in terms of performance.

### CODE REVIEW

**Comment by merav-aharoni** [comment]
@atilag , @chris, this PR is ready for review.

File: src/simulators/tensor_network/matrix_product_state.cpp [hunk 1]
```diff
@@ -21,6 +21,8 @@
 #include "matrix_product_state.hpp"
 #include "matrix_product_state_tensor.hpp"
 
+using namespace std;
```
Comment (chriseclectic) [inline src/simulators/tensor_network/matrix_product_state.cpp]: Can you remove `using namespace std;`

**Comment by chriseclectic** [comment]
@merav-aharoni I just noticed that some of the mps files are missing the qiskit copyright header. Can you fix it for `svd.cpp`, `svd.hpp`, and `matrix_product_state_tensor.hpp`

File: src/simulators/tensor_network/matrix_product_state.cpp (hunk 1 above)
Comment (merav-aharoni) [reply src/simulators/tensor_network/matrix_product_state.cpp]: Done. Also removed from svd.hpp and svd.cpp

**Comment by chriseclectic** [comment]
@merav-aharoni does this need rebasing after the recent MPS fixes?

**Comment by merav-aharoni** [comment]
Hi @chriseclectic , I merged with master. You can continue to review now.

**Comment by merav-aharoni** [comment]
@chriseclectic I believe this PR can be merged.

File: src/simulators/matrix_product_state/svd.cpp [hunk 2]
```diff
@@ -452,17 +451,17 @@ status csvd(cmatrix_t &A, cmatrix_t &U,rvector_t &S,cmatrix_t &V)
 	{
 		if ( b[k] != 0.0 )
```
Comment (chriseclectic) [inline src/simulators/matrix_product_state/svd.cpp]: `!Linalg::almost_equal(b[k], 0.0)`

File: src/simulators/matrix_product_state/svd.cpp [hunk 3]
```diff
@@ -475,7 +474,7 @@ status csvd(cmatrix_t &A, cmatrix_t &U,rvector_t &S,cmatrix_t &V)
 		k1 = k + 1;
 		if ( c[k1] != 0.0 )
```
Comment (chriseclectic) [inline src/simulators/matrix_product_state/svd.cpp]: `!Linalg::almost_equal(c[k1], 0.0)`

File: CHANGELOG.md [hunk 4]
```diff
@@ -48,7 +48,7 @@ Removed
 Fixed
 -----
 - Fix sdist to always attempt to build (\#401)
-
+- New (efficient) implementation for expectation_value_pauli (\#344)
```
[the remark this answers is not in the document]
Comment (merav-aharoni) [reply CHANGELOG.md]: Done (actually that was already the name before your comment :-) )

File: src/simulators/matrix_product_state/svd.cpp [hunk 5]
```diff
@@ -178,9 +177,9 @@ status csvd(cmatrix_t &A, cmatrix_t &U,rvector_t &S,cmatrix_t &V)
 
 		if ( tol < z )
 		{
-			z = sqrt( z );
+			z = std::sqrt( z );
 			c[k1] = z;
-			w = abs( A(k,k1) );
+			w = std::abs( A(k,k1) );
 
 			if ( w == 0.0 ){
```
[the remark this answers is not in the document]
Comment (merav-aharoni) [reply src/simulators/matrix_product_state/svd.cpp]: @chriseclectic I agree with your comments on replacing `==` with `Linalg::almost_equal`.
However, this is unrelated to the code for the expectation value, so I prefer to handle as a separate issue. I opened a new issue #468 for this purpose.
Please let me know if you have any additional comments for this PR.

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chriseclectic) [reply CHANGELOG.md]: I meant can you change the line in the changelog to match the new title for this PR.

**Comment by merav-aharoni** [comment]
@chriseclectic , Changed the comment in CHANGELOG.md as you suggested.

**Pull request merged**